### DISCUSSION TITLE
Nested navigation

### DISCUSSION THREAD

**Discussion by Evoolo** [scalar/scalar#1016] [Ideas]
Hi,

Is there a good way to create a nested navigation? I want to use scalar with scribe but i won't miss the subgroups feature.

My idea was to create a new configuration parameter in both, scribe and scalar. In scribe the new flag would allow to export the subgroups as a second tag in the openapi yaml file.

In scalar the new configuration parameter would toggle the subgroups feature. If turned on, the second tag will be usded as the title of the nested navigation.

Please let me know what you think about that feature an I will try to add a PR

**Comment by hanspagel** [comment]
Oh, cool! Can you share a screenshot of how subgroups look like?

**Reply by Evoolo** [reply]
Sure, they look like this in the elements theme for example

![Screenshot_2024-02-17-00-00-53-871_org.mozilla.firefox-edit.jpg](https://github.com/scalar/scalar/assets/21312800/3109f71a-df19-4b26-a278-ba2e1aaf7d21)

**Reply by hanspagel** [reply]
Thanks so much! I’m bit hesitant if it’s not part of the OpenAPI standard, but there might be an even better way to leverage the standard. I need to think about that!

**Reply by Evoolo** [reply]
Great to hear that! I know it's not standard Just come back for discussions. But I have about 100 requests and the sidebar is looking loaded. Just come back if you want to discuss it further.

**Comment by noxasch** [comment]
redoc is using custom attribute for this [x-tag-groups](https://redocly.com/docs/api-reference-docs/specification-extensions/x-tag-groups/) since it's not yet implement in open api spec.

**Comment by marclave** [comment]
@noxasch @Evoolo we implemented `x-tagGroups` in this PR :) just merged!

https://github.com/scalar/scalar/pull/1174

released in version `1.19.3` ✨

**Reply by Evoolo** [reply]
This are great news! 👏

I will take a look on it next week. 

Thank you and keep up the great work.

**Reply by noxasch** [reply]
Awesome job @marclave , I tested it, however models is missing from the sidebar menu if I set `x-taggroups`, is this to be expected ?

**Reply by marclave** [reply]
yes this is aligned with how x-taggroups works, if you want models to show up you can add it as a taggroup :)

**Reply by Evoolo** [reply]
@noxasch can you share your openapi file?

**Reply by noxasch** [reply]
@Evoolo I'm using swagger for ruby, rswag. My swagger/open-api spec looks something like this when using `x-tagGroups`, but couldn't figure out how to use models @marclave if you have any clue, let me know

```yaml
---
openapi: 3.0.1
info:
  title: My App API V1
  version: v1
  description: This is the backend api documentation. Select you preferred language
    library and you can set your bearer token first before test the api.
x-tagGroups:
- name: Oauth
  tags:
  - Oauth
- name: Users
  tags:
  - Transaction
  - Pocket
paths:
  "/api/v1/users/transactions":
    get:
      summary: Retrieve transactions
      tags:
      - Transaction
      description: Transaction
      security:
      - bearerAuth: []
      responses:
        '200':
           description: Success
```

**Comment by marclave** [comment]
@Evoolo @noxasch ah good catch! I totally made a mistake there, PR coming up soon :) 

https://github.com/scalar/scalar/pull/1305
